scripts/get-prs

```diff
@@ -20,13 +36,93 @@ import sys
+
+# Now things get a bit tricky, Both "pr_ref_short_re" and "pr_ref_url_re" are
+# atoms, but they have matching group inside. When or'ing multiple regex'es,
+# each sub-regex contribute the groups, even if they did not match. Thus, any
+# match of "pr_ref" will yield a 3-tupel, where either the first element or the
+# last two elements ("org_re/repo_re" and "pr_id_re") are not empty. That nicely
+# allows determining what PR reference form we have found.
+pr_ref_short_re = regex_atom(f"({org_re}/{repo_re}#{pr_id_re})")
+pr_ref_url_re = regex_atom(f"https://github.com/({org_re}/{repo_re})/pull/({pr_id_re})")
+pr_ref = regex_atom(f"{pr_ref_short_re}|{pr_ref_url_re}")
+
+# Since a test can run with PRs from multiple repos, we have to support lists
+# also. A hacky way to achieve this is just defining an atom that is a list
+# item with optionally another separator-atom attached - so a list is just a
+# sequence of such items. That is good enough practically.
+sep_comma_re    = regex_atom(f"{whitespace_re}*,{whitespace_re}*")
+sep_and_re      = regex_atom(f"{whitespace_re}+and{whitespace_re}+")
+sep_line_end_re = regex_atom(f"{whitespace_re}*$")
+repo_sep_re     = regex_atom(f"{sep_comma_re}|{sep_and_re}|{sep_line_end_re}")
+pr_list_item_re = regex_atom(f"{pr_ref}{repo_sep_re}")
+
+# Finally, the complete regex can be build. The "$" in the end ensures only
+# lines without any trailing data are accepted.
+# Unfortunately, in Python the number of groups that match an expression are
+# static, for repeating matches like in the list, only the last match is
+# returned for each group. To work around this, we have to make "pr_list" a
+# matching group here and then find all matching "pr_ref" in a second step.
+test_with_prs_re   = regex_atom(f"[Tt]est with:?{whitespace_re}*({pr_list_item_re}+)$")
+
+# Build the compiled regex'es that will be used
+test_with_prs_cre = re.compile(test_with_prs_re)
+pr_list_item_cre = re.compile(pr_list_item_re)
+
+
+def find_pr_to_test_with(text):
+  prs = []
+  if text is not None:
+    for line in text.splitlines():
+      m = test_with_prs_cre.match(line)
+      if m:
+          list_pr_list = m.group(1)
+          for t in pr_list_item_cre.findall(list_pr_list):
+            #print("  PR:", t)
+            prs.append(t[0] if t[0] else f"{t[1]}#{t[2]}")
+
+  return prs
+
+def unit_test():
+    a = find_pr_to_test_with(
+        "This is a test\n" \
+        "Test with: org1/repo1#41\n" \
+        "test with https://github.com/org2/repo2/pull/42\n" \
+        "Test with: \n" \
+        "Test with:  org3/repo3#43 and org4/repo4#44,https://github.com/org5/repo5/pull/45\n" \
+        "-- Trailing list separators are accepted:\n" \
+        "Test with: https://github.com/org6/repo6/pull/46,\n" \
+        "Test with: org7/repo7#47 and \n" \
+        "-- There must be a valid separator\n" \
+        "Test with: org91/repo91#91org92/repo92#92\n" \
+        "Test with: org93/repo93#93 org94/repo94#94\n" \
```

Comment:
Do you really want `org91/repo91#91org92/repo92#92` to pass also, so it yields both `org91/repo91#91` and `org92/repo92#92`? Since we match the expression based on a very strict pattern, that would work actually - but I think it'sgood practice to require at least some separator there.

Reply:
I wouldn't want anyone to actually write that, but I wouldn't mind if it was matched. AFAICS it would always match correctly, even with the URL extension, because we always have a number at the end.

The main thing I want is support for comments after the expression, so `$` is definitely out.

We could also just do it properly and remove my hack, i.e. have something like `<pr_ref> (<sep>[ ]+<pr_ref>)*` as the expression and a bit of extra handling to return one list in the end.

Reply:
> We could also just do it properly and remove my hack

Done it this was. This accepts the "1xx" references, but ignores the "9xx" references then:
```
        "-- A proper separator is required, otherwise it's considered a comment\n" \
        "-- This can become quite odd\n" \
        "Test with: org114/repo114#114org902/repo902#902\n" \
        "Test with: org115/repo115#115 org116/repo116#116org903/repo903#903\n" \
        "Test with: org117/repo117#117andorg904/repo904#904\n" \
        "Test with: org118/repo118#118gliberishorg905/repo905#905\n" \
```